From 53a6c1a7cb4a4db334c794fde1ea459a67df28a6 Mon Sep 17 00:00:00 2001
From: Liang Li <liang.li@windriver.com>
Date: Mon, 23 Aug 2010 14:36:33 +0800
Subject: [PATCH 2/2] wrhv/powerpc: switch for paravirt dcache_clean

Add a macro switch to indicate there is no need to paravirt
dcache_clean_range function but currently we will just use
it. The new macro switch could remind us that we can try
native dcache range later.

Signed-off-by: Liang Li <liang.li@windriver.com>
---
 arch/powerpc/include/asm/wrhv.h    |   11 +++++++++++
 arch/powerpc/kernel/wrhv_misc_32.S |    8 +++++---
 2 files changed, 16 insertions(+), 3 deletions(-)

diff --git a/arch/powerpc/include/asm/wrhv.h b/arch/powerpc/include/asm/wrhv.h
index aea8245..8208029 100644
--- a/arch/powerpc/include/asm/wrhv.h
+++ b/arch/powerpc/include/asm/wrhv.h
@@ -16,6 +16,16 @@
 #define __ASM_WRHV_H
 
 #ifdef CONFIG_WRHV
+
+/*
+ * This macro will tend to clean_dcache_range function, but
+ * we do not need to do that, so delete below macro to try
+ * native clean_dcache_range
+ */
+#define CONFIG_PARAVIRT_DCACHE_CLEAN
+
+#ifndef __ASSEMBLY__
+
 extern unsigned char *law_base;
 
 extern void wrhv_mapping(void);
@@ -49,5 +59,6 @@ extern int ppc_get_pci_intr_wrhv(struct pci_dev *dev);
 extern int wrhv_enable_pci_law(void);
 #endif
 
+#endif /* __ASSEMBLY__ */
 #endif /* CONFIG_WRHV */
 #endif /* __ASM_WRHV_H */
diff --git a/arch/powerpc/kernel/wrhv_misc_32.S b/arch/powerpc/kernel/wrhv_misc_32.S
index 0bbec39..79a84b3 100644
--- a/arch/powerpc/kernel/wrhv_misc_32.S
+++ b/arch/powerpc/kernel/wrhv_misc_32.S
@@ -24,6 +24,7 @@
 #include <asm/processor.h>
 #include <vbi/interface.h>
 #include <vbi/syscalls.h>
+#include <asm/wrhv.h>
 
 	.text
 
@@ -42,11 +43,11 @@ _GLOBAL(wrhv_int_unlock)
 
 /*
  * Write any modified data cache blocks out to memory.
- * Does not invalidate the corresponding cache lines (especially for
- * any corresponding instruction cache).
+ * Does not invalidate the corresponding data cache lines
  *
- * clean_dcache_range(unsigned long start, unsigned long stop)
+ * paravirt_clean_dcache_range(unsigned long start, unsigned long stop)
  */
+#ifdef CONFIG_PARAVIRT_DCACHE_CLEAN
 _GLOBAL(paravirt_clean_dcache_range)
 	/*
 	 * vbi_flush_dcache (void *start_addr, void *end_addr)
@@ -56,6 +57,7 @@ _GLOBAL(paravirt_clean_dcache_range)
 	addi	r4, r4, 1
 	bl	vbi_flush_dcache
 	blr
+#endif
 
 _GLOBAL(paravirt__flush_dcache_icache)
 	b	vb__flush_dcache_icache
-- 
1.6.5.2

